test-data/unit/check-expressions.test

```diff
@@ -936,8 +936,8 @@ main:3: error: "A" not callable
 
 a, o = None, None # type: (A, object)
 a = f()         # E: "f" does not return a value
-o = a()         # E: Function does not return a value
-o = A().g(a)    # E: "g" of "A" does not return a value
+o = a()
+o = A().g(a)
```

Comment:
I think that this should still generate an error, since `A.g` has an explicit `None` return type.